File: 2025/caju.cpp
#include<iostream>
using namespace std;

int L, C, M, N;

int main(void) {
  int max = 0;
  cin >> L >> C >> M >> N;
  int matriz[L+2][C+2];
  int acumulado[L+2][C+2];
  for(int i = 1; i <= L; i++){
    for(int j = 1; j <= C; j++){
      cin >> matriz[i][j];
    }
  }
  for(int i = 0; i <= L; i++){
    acumulado[i][0] = 0;
    acumulado[0][i] = 0;
  }
  for(int i = 1; i <= L; i++){
    for(int j = 1; j <= C; j++){
      acumulado[i][j] = acumulado[i-1][j] + acumulado[i][j-1]- acumulado[i-1][j-1] + matriz[i][j];
    }
  }


  for (int i = M; i <= L; i++) {
    for (int j = N; j <= C; j++) {
      int ponto = acumulado[i][j] - acumulado[i - M][j] - acumulado[i][j - N] + acumulado[i - M][j - N];
      if (ponto > max) {
        max = ponto;
      }
    }
  }

  int x;
  cout << max << endl;



  return 0;
}

File: 2025/caju.py
import sys


L=eval(input(""))
C=eval(input(""))
M=eval(input(""))
N=eval(input(""))
matriz = []
acumulada = []
max = 0

for i in range(L+1):
    matriz.append([])
    acumulada.append([])

for i in range(L+1):
    matriz[i].append(0)
    acumulada[i].append(0)

for i in range(C):
    matriz[0].append(0)
    acumulada[0].append(0)
for i in range(1,L+1):
    for j in range(1,C+1):
        matriz[i].append(eval(input("")))
        acumulada[i].append(0)

for i in range(1,L+1):
    for j in range(1,C+1):
        acumulada[i][j] = acumulada[i-1][j] + acumulada[i][j-1] - acumulada[i-1][j-1] + matriz[i][j]
for i in range(M,L+1):
    for j in range(N,C+1):
        pontos = acumulada[i][j] - acumulada[i-M][j] - acumulada[i][j-N] + acumulada[i-M][j-N]
        if(pontos>max):
            max = pontos
print(max)

sys.exit(0)  